lib/active_merchant/billing/gateways/decidir.rb

```diff
@@ -266,6 +266,11 @@ def error_code_from(response)
           error_code ||= error['type']
         elsif response['error_type']
           error_code = response['error_type'] if response['validation_errors']
+        elsif error = response.dig('error')
+          validation_errors = error.dig('validation_errors', 0)
```

Comment:
Do we need to provide 0 here? Wouldn't `validation_errors` be set to nil otherwise?

Reply:
So, the response that I'm working from is this:
`{\"error\":{\"error_type\":\"invalid_request_error\",\"validation_errors\":[{\"code\":\"invalid_param\",\"param\":\"payment_type\"}]}}`
And that ends up being a nested array. I'm only grabbing it at index 0 because I'm not sure what a response with additional indexes would look like.

Reply:
Ohhhh I misunderstood what was going on here. Seems sensible 👍